File: 231-Power-of-Two/CPP/Solution02.cpp
// ==============================================================================
// Author:   Tao Li (taoli@ucsd.edu)
// Date:     Jul 7, 2015
// Question: 231-Power-of-Two
// Link:     https://leetcode.com/problems/power-of-two/
// ==============================================================================
// Given an integer, write a function to determine if it is a power of two.
// ==============================================================================
// Method: loop
// ==============================================================================

class Solution {
public:
    bool isPowerOfTwo(int n) {
        if (n <= 0) return false;
        while (n >= 2) {
            if (n % 2 == 1) return false;
            n /= 2;
        }
        return true;
    }
};

int main(int argc, char const *argv[])
{
    /* code */
    return 0;
}

File: 231-Power-of-Two/Python/Solution02.py
#!/usr/bin/python
"""
==============================================================================
Author:   Tao Li (taoli@ucsd.edu)
Date:     Jul 7, 2015
Question: 231-Power-of-Two
Link:     https://leetcode.com/problems/power-of-two/
==============================================================================
Given an integer, write a function to determine if it is a power of two.
==============================================================================
Method: loop
==============================================================================
"""

class Solution:
    # @param {integer} n
    # @return {boolean}
    def isPowerOfTwo(self, n):
        if n <= 0:
            return False

        while n >= 2:
            if n % 2 == 1:
                return False
            n /= 2

        return True
